### ISSUE TITLE
Combination of Data Sampler and Predictions produces incorrect output when opening workflow from saved file

### ISSUE THREAD

**Issue by wvdvegte** [biolab/orange3#5989]
**What's wrong?**
When using Data Sampler to split a dataset in a training set (n%) and a test set ((100-n)%), with Predictions connected to the Remaining Data output (test set)  and a trained model, Predictions will send only one prediction on its output instead of predictions for (100-n)% of the dataset. Inits own dialog box, Predictions seems to show the correct number of predictions

**How can we reproduce the problem?** 
Open the attached workflow, let it load, and check the Data Table on the output of Predictions. There is only one row with one prediction. Now ope the Data Sampler Widget, press the Sample Data button, and observe that the output of Predictions does show the correct number of predictions

[sampling test.ows.zip](https://github.com/biolab/orange3/files/8756142/sampling.test.ows.zip)

**What's your environment?**

- Operating system: Mac OS 11.6.5 Intel
- Orange version: 3.32.0
- How you installed Orange: fom DMG

**Comment by janezd** [comment]
Sample Data outputs 99 instances, which are shown in Predictions. In Predictions, you selected 1 data instance, which is shown in Table.

If you click Sample, Sample data re-outputs the same 99 instances, which are again shown in Predictions. However, Predictions reset the selection (i.e. nothing is selected) and outputs the entire data set.

If you deselect the selected data instance in Predictions (instead of clicking Sample), the Table will also show all 99 instances.

Works as intended.

**Comment by wvdvegte** [comment]
Thanks for the explanation. I wasn't aware that it is possible to select one or more rows in Predictions, and apparently made the selection unconsciously.

But IMO this means that the output of Predictions should offer a choice between "Selected Predictions" and "All Predictions". That would be similar to the output options of Data Table and therefore more consistent. If the connector from Predictions to Data Table  would have been labeled "Selected Predictions", it would immediately have been clear to me what was happening. But then again, I would prefer that the default setting in this case would be "All Predictions". I see no reason to make selections in Predictions: one might as well do that in a downstream Data Table.

And, by the way, it is also quite hard to find out how to deselect in Predictions. Apparently, this can only be done by clicking in the white area to the right of the table.
